### PR TITLE
React static for Static Site Generation

### PR DESCRIPTION
**Pull request by Apoorv-Mittal**
As discussed in #8 I have tried to implement react static to render static pages of the website. This is my first thing using React Static so any feedback is welcomed and appreciated.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "react-static/.babelrc",
+  "plugins": ["react-hot-loader/babel"]
```
Comment (philipp-spiess) [inline .babelrc]: I don't think we need hot reloading. Can we remove this?

File: package.json [hunk 2]
```diff
@@ -1,8 +1,31 @@
...
+  "license": "MIT",
+  "scripts": {
+    "start": "react-static start",
+    "stage": "react-static build --staging",
+    "build": "react-static build",
+    "serve": "serve dist -p 3000"
+  },
+  "dependencies": {
+    "chokidar": "2.0.4",
+    "htmr": "0.7.0",
```
Comment (philipp-spiess) [inline package.json]: What is this dependency used for? I don't see it referenced anywhere.

File: package.json [hunk 3]
```diff
@@ -1,8 +1,31 @@
...
+    "jdown": "^0.6.5",
```
Comment (philipp-spiess) [inline package.json]: The same goes for this dependency.

File: src/containers/Home.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+              <meta name="twitter:site" content="@PhilippSpiess" />
+              <meta name="twitter:title" content="This Week In React ⚛️" />
+              <meta name="twitter:description" content="Weekly coverage of interesting pull requests, discussions, and RFCs around React Core and React DOM." />
+              <meta name="twitter:image" content="https://philippspiess.com/this-week-in-react/twitter-card.png" />
+              <link href="https://fonts.googleapis.com/css?family=Roboto:300,700" rel="stylesheet" />
+            </Helmet>
+            <div aria-hidden="true" className="atoms" />
+              <div className="main">
+                <section>
+                  <h1>This Week In React <img src={ ReactSVG } width= "60px"  alt="⚛️" id="logo" /></h1>
```
Comment (philipp-spiess) [inline src/containers/Home.js]: Can you add an onClick event to the `` that also calls `createBackgroundPattern()`? This way we keep the current behavior of the easter egg 🙂

File: src/containers/atoms.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  const neighborhood = 150;
+  let i = 0;
+  for (let x = 0; x < window.screen.width; x += neighborhood) {
+    for (let y = 0; y < window.screen.height; y += neighborhood) {
+      createRandomAtom(i, x, y);
+      i += 1;
+    }
+  }
+}
+// document.getElementById("logo").onclick = createBackgroundPattern;
```
Comment (philipp-spiess) [inline src/containers/atoms.js]: This can be removed

File: static.config.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { reloadRoutes } from 'react-static/node'
+import chokidar from 'chokidar'
+
+
+chokidar.watch('issue').on('all', () => reloadRoutes())
+
+export default {
+  getSiteData: () => ({
+    title: 'React Static',
```
Comment (philipp-spiess) [inline static.config.js]: This should probably return "This Week In React"

File: static.config.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import { reloadRoutes } from 'react-static/node'
+import chokidar from 'chokidar'
+
+
+chokidar.watch('issue').on('all', () => reloadRoutes())
```
Comment (philipp-spiess) [inline static.config.js]: Can we remove this for now? I assume we don't necessarily need it

**Review by philipp-spiess — COMMENTED** [review]
This is great! The static compilation also works as I expect it to.

The favicon is not working anymore - I think you have to move it to `/public` as well.

When you're done with the changes I'll look into how I can upgrade my GitHub pages integration so that it properly uses the files that I create via `yarn build`. [This](https://github.com/ninaolo/react-static-github-pages-example) should be helpful!

File: package.json (hunk 2 above)
Comment (Apoorv-Mittal) [reply package.json]: It converts html to react, which I was using before but I removed that. I forgot to remove it from the package file. I will remove it in the next PR.

File: .babelrc (hunk 1 above)
Comment (Apoorv-Mittal) [reply .babelrc]: I will try and remove it

File: package.json (hunk 3 above)
Comment (Apoorv-Mittal) [reply package.json]: I was using it to convert a directory of markdown to html but the repo does not need it now. I will remove it.

File: src/containers/Home.js (hunk 4 above)
Comment (Apoorv-Mittal) [reply src/containers/Home.js]: 👍

File: src/containers/atoms.js (hunk 5 above)
Comment (Apoorv-Mittal) [reply src/containers/atoms.js]: 👍

**Review by philipp-spiess — APPROVED** [review]
Thank you for all your work - This PR is great! ❤️ 

I've made some changes that are more or less nitpicks. Unfortunately I found what appears to be a bug in react-static (https://github.com/nozzle/react-static/issues/816). We'll see if they respond quickly otherwise I'll have to look into a workaround - Serving unminified React is not a great idea for initial page load time 🙂 

I've yet to set it up so that it builds the production artifacts into a different GitHub branch but that shouldn't be too hard.

Oh and sorry for the long waiting time - Thinks are unfortunately pretty busy on my side.

**Comment by philipp-spiess** [comment]
@Apoorv-Mittal This issue is finally resolved 🎉 It turns out that the yarn lockfile had an issue and that the `performance-now` package was having the contents of a completely different package. Probably caused by a merge issue?

Anyway the new react-static powered website is already online. Thank you so much for your help. 🙌

**Pull request merged**